tools/schemacode/bidsschematools/render.py

```diff
@@ -469,11 +469,56 @@ def make_filename_template(
         )
 
     codeblock = codeblock.expandtabs(4)
+    codeblock = append_filename_template_legend(codeblock, pdf_format)
     codeblock = codeblock.replace("SPEC_ROOT", get_relpath(src_path))
 
     return codeblock
 
 
+def append_filename_template_legend(
+    text=None,
+    pdf_format=False,
+):
+
+    if text is None:
+        return
+
+    legend = """
+<li>Filename entities or folders between square brackets
+    (for example <code>[_ses-<label>]</code>) are OPTIONAL.</li>
+<li>Some entities may only allow specific values,
+    in which case those values are listed in <code><></code>, separated by <code>|</code>.</li>
+<li><code>_<suffix></code> means that there are several (>6) valid suffixes
+    for this filename pattern.</li>
+<li><code>.<extension></code> means that there are several (>6) valid extensions
```

Comment:
```suggestion
<li><code>_&lt;suffix&gt;</code> means that there are several (>6) valid suffixes
    for this filename pattern.</li>
<li><code>.&lt;extension&gt;</code> means that there are several (>6) valid extensions
```

Reply:
👍 for this proposed change. The `_<suffix>` and `.<extension>` elements currently show up as `_` and `.`.

Reply:
Do we just want to do our own Markdown render? e.g., write this as:

```
- `_<suffix>` ...
```

And then something like `html = commonmark.render(md)`?

Reply:
> Do we just want to do our own Markdown render?

you mean as part of this PR? 
(sorry I am very dense today)

Reply:
Either way. It just looks like you're manually writing HTML, and then modifying it for the benefit of the LaTeX renderer back to something closer to Markdown. And we're hitting a case where angle brackets `<>` are hiding things in HTML when converting them to HTML escape characters is almost certainly going to mess up the TeX. I guess I'm just wondering if it's worth doing so much by hand.

Reply:
I used the successor to `commonmark` to do the job